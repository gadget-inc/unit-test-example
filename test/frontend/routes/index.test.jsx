import { describe, expect, it } from "vitest";
import renderer from "react-test-renderer";
import Index from "../../../frontend/routes/index";

// a simple example of a snapshot test
// this is similar to testing basic React components that do not require mocks
describe("snapshot test for frontend/routes/index.jsx", () => {
  it("should render the Index component", () => {
    const tree = renderer.create(<Index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
