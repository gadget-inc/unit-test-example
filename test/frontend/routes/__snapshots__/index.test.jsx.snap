// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test for frontend/routes/index.jsx > should render the Index component 1`] = `
[
  <div
    className="app-link"
  >
    <img
      alt="logo"
      className="app-logo"
      src="/frontend/assets/react-logo.svg"
    />
    <span>
      You are now signed out of 
       Â 
    </span>
  </div>,
  <div>
    <p
      className="description"
    >
      Start building your app's signed out area
    </p>
    <a
      href="/edit/files/frontend/routes/index.jsx"
      rel="noreferrer"
      style={
        {
          "fontWeight": 500,
        }
      }
      target="_blank"
    >
      frontend/routes/index.jsx
    </a>
  </div>,
]
`;
